MyThreadPool/MyThreadPool/MyThreadPool.cs

```diff
@@ -0,0 +1,204 @@
+﻿namespace MyThreadPool;
+
+/// <summary>
+/// MyTPL
+/// </summary>
+public class MyThreadPool
+{
+    private int NumberOfThreads;
```

Comment:
```suggestion
    private int numberOfThreads;
```
Поля в .NET всегда именуются со строчной. Настройте себе StyleCop